In [23]:
import numpy as np
import scipy.stats as stats

## Задача 1

## Задача 2

### Формируем гипотезы

In [22]:
#т.к. дисперсия известна, принимаем для расчета Z-критерий
zn = (17.5-17)/(np.sqrt(4)/np.sqrt(100))
print(zn)

2.5


In [4]:
#найдем z по таблице z-значений для критерия 1-a, чтобы сравнить z и zn
z = (1.64+1.65)/2
print(z)

1.645


## Задача 3

### Формируем гипотезы

In [15]:
#т.к. дано нормальное распределение и дисперсия выборки нам не изместна, применяем критерий Стьюдента
cookies = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
xm = cookies.mean()
sigma = cookies.std(ddof=1)
t = (xm-200)/(sigma/np.sqrt(10))
print(t)

-1.0651074037450896


In [19]:
#табличное значение критерия Стьюдента
a = (100-99)/100/2 ##т.к. тест двусторонний
tt = 3.25 #(для значения выборки 10-1 = 9)

In [20]:
#т.к. анализ двусторонний, возьмем абсолютное значение t и сравним его с табличным
np.abs(t)>tt

False

## Задача 4

In [ ]:
Есть ли статистически значимые различия в росте дочерей?
Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165
Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160, 163

### Формируем гипотезы

In [ ]:
Гипотеза H0: статистически значимые различия в росте дочерей отсутствуют
Гипотеза H1: статистически значимые различия в росте дочерей есть

In [25]:
mothers = np.array([172, 177, 158, 170, 178, 175, 164, 160, 169, 165])
daughters = np.array([173, 175, 162, 174, 175, 168, 155, 170, 160, 163])

In [28]:
print(stats.ttest_rel(mothers, daughters))

Ttest_relResult(statistic=0.664847853143198, pvalue=0.5228168632983574)


Вывод: т.к. pvalue = 0.523 больше минимального значения a равного 0.05, то гипотезу H1 отвергаем. Следовательно, нет статистически значимых различий в росте дочерей